test/SparqlParser-test.js

```diff
@@ -37,6 +37,15 @@ describe('A SPARQL parser', function () {
     expect(error.message).to.include('Parse error on line 1');
   });
 
+  it('should preserve BGP and filter pattern order', function () {
+    var parser = new SparqlParser();
+    var query = 'SELECT * { ?s ?p "1" . FILTER(true) . ?s ?p "2"  }';
```

Comment:
This is actually a case where order doesn't matter, but I suppose SPARQL.js shouldn't mess with the order in any case. Query optimizers can always later move things around.